packages/view/backend/sirius-components-view-emf/src/main/java/org/eclipse/sirius/components/view/emf/diagram/ViewDiagramDescriptionConverter.java

```diff
@@ -635,16 +642,39 @@ private EdgeDescription convert(org.eclipse.sirius.components.view.EdgeDescripti
 
     private Function<VariableManager, IStatus> createDeleteHandler(DiagramElementDescription diagramElementDescription, ViewDiagramDescriptionConverterContext converterContext) {
         var capturedConvertedNodes = Map.copyOf(converterContext.getConvertedNodes());
-        DeleteTool tool = diagramElementDescription.getDeleteTool();
-        if (tool != null) {
-            return variableManager -> {
+        return variableManager -> {
+            IStatus result;
+            DeletionPolicy deletionPolicy = variableManager.get(DeleteFromDiagramEventHandler.DELETION_POLICY, DeletionPolicy.class).orElse(DeletionPolicy.SEMANTIC);
+            if (deletionPolicy == DeletionPolicy.GRAPHICAL) {
+                this.deleteFromDiagram(variableManager);
+                result = new Success();
+            }
+            DeleteTool tool = diagramElementDescription.getDeleteTool();
+            if (tool != null) {
                 VariableManager child = variableManager.createChild();
                 child.put(CONVERTED_NODES_VARIABLE, capturedConvertedNodes);
-                return new DiagramOperationInterpreter(converterContext.getInterpreter(), this.objectService, this.editService, this.getDiagramContext(variableManager), capturedConvertedNodes)
+                result = new DiagramOperationInterpreter(converterContext.getInterpreter(), this.objectService, this.editService, this.getDiagramContext(variableManager), capturedConvertedNodes)
                         .executeTool(tool, child);
-            };
-        } else {
-            return this.canonicalBehaviors::deleteElement;
+            } else {
+                result = this.canonicalBehaviors.deleteElement(variableManager);
+            }
```

Comment:
This does not work. There is an `else` missing since it will otherwise perform both the graphical deletion and the semantic deletion too.